src/ncp/ncp_base_ftd.cpp

```diff
@@ -1323,7 +1323,9 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_CHANNEL_MANAGER_CHANN
     otError error            = OT_ERROR_NONE;
 
     SuccessOrExit(error = mDecoder.ReadBool(skipQualityCheck));
+#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
```

Comment:
The same suggestion for the related spinel properties. 

I would suggest `SPINEL_PROP_CHANNEL_MANAGER_CHANNEL_SELECT` to  be provided when `CHANNEL_MONITOR` (and `CHANNEL_MANAGER`) featuers are enabled. Keeping the property that does nothing (when feature is disabled) does not sound good.